### PR TITLE
powerdns driver adapted to work with pdns 3 and 4

### PR DESCRIPTION
**Pull request by biggosh**
### Description

Added support for PowerDNS API version 1 maintaining the backward compatibility with experimental API version.

### CODE REVIEW

**Comment by biggosh** [comment]
Hi. I've checked the error, but I can't understand the reason.

File: libcloud/dns/drivers/powerdns.py [hunk 1]
```diff
@@ -143,7 +144,7 @@ def __init__(self, key, secret=None, secure=False, host=None, port=None,
                                              host=host, port=port,
                                              **kwargs)
 
-    def create_record(self, name, zone, type, data, extra=None):
+    def create_record(self, name, zone, type, data, extra=None, disabled=False, comment=None):
```
Comment (Kami) [inline libcloud/dns/drivers/powerdns.py]: Those are extension arguments (arguments which are not supported by all the providers), please prefix them with ``ex_``.

So ``ex_disabled`` and ``ex_comment``.

File: libcloud/dns/drivers/powerdns.py [hunk 2]
```diff
@@ -368,7 +392,7 @@ def list_zones(self):
         response = self.connection.request(action=action, method='GET')
         return self._to_zones(response)
 
-    def update_record(self, record, name, type, data, extra=None):
+    def update_record(self, record, name, type, data, extra=None, disabled=False, comment=None):
```
Comment (Kami) [inline libcloud/dns/drivers/powerdns.py]: Same here.

**Comment by Kami** [comment]
Thanks for the contribution.

It would also be good to add some test cases which test both API version - 3 and 4.

File: libcloud/dns/drivers/powerdns.py [hunk 3]
```diff
@@ -439,13 +485,27 @@ def _to_zones(self, items):
...
+            for item in items.object['records']:
+                records.append(self._to_record(item, zone))
+        if self._pdns_version() == 4:
+            for item in items.object['rrsets']:
+                for record in item['records']:
+                    records.append(self._to_record(item, zone, record))
         return records
+		
+    def _pdns_version(self):
+        if self.api_root == '': return 3
```
Comment (Kami) [inline libcloud/dns/drivers/powerdns.py:515]: How is version currently declared? To which version does the driver currently default to?

Comment (Kami) [reply libcloud/dns/drivers/powerdns.py:515]: EDIT: I see it's declared here - https://github.com/apache/libcloud/blob/9cab4689ac005fed2de0761174cf13981d525128/libcloud/dns/drivers/powerdns.py#L133

File: libcloud/dns/drivers/powerdns.py (hunk 2 above)
Comment (biggosh) [reply libcloud/dns/drivers/powerdns.py]: The new push has been modified: disabled and comment has been added to extra parameter already declared

**Comment by biggosh** [comment]
I cannot add tests because I don't know how to add (I'm a very old school programmer) them and I'm programming also in free time, just not to forgot my first IT love. Sorry

**Comment by Kami** [comment]
I made some styling changes to get build to pass and merged this into trunk.

Thanks again for your contribution. In the future, if you get a chance, it would be good to add some tests for this functionality.

**Pull request merged**